### PR TITLE
Feature/project api

### PR DESCRIPTION
**Pull request by willyyto**

### CODE REVIEW

File: src/server/StudioManagementSystem.Infrastructure/Repositories/GroupRepositoryAsync.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        var myGroup = await _smsDbContext.Groups.FindAsync(id, ct);
+        try
+        {
+            myGroup.Name = group.Name;
+            await _smsDbContext.SaveChangesAsync(ct);
+        }
+        catch (Exception ex)
+        {
+            return null;
+        }
```
Comment (albert118) [inline src/server/StudioManagementSystem.Infrastructure/Repositories/GroupRepositoryAsync.cs:56]: the find async may throw (same with the method that repeats this logic). Let's re-use the existing method, check we got the group, and add some logging

```suggestion
        var group = await GetGroupAsync(id, ct)
            ?? throw new DataException($"No {nameof(Group)} was found with the given ID '{id}'");

        try
        {
            group.Name = name;
            await _smsDbContext.SaveChangesAsync(ct);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, $`an error occurred while updating the {nameof(Group)} name`)
            return false;
        }
```

Comment (albert118) [reply src/server/StudioManagementSystem.Infrastructure/Repositories/GroupRepositoryAsync.cs:56]: This means if we screw up in the rest of the app we'll quickly know,

* did we remember the ID?
* OR what was the unknown error we encountered?

Comment (albert118) [reply src/server/StudioManagementSystem.Infrastructure/Repositories/GroupRepositoryAsync.cs:56]: I'd repeat this logging pattern in particular around the repo's

File: src/server/StudioManagementSystem/DependencyInjectionExtensions.cs [hunk 2]
```diff
@@ -86,6 +87,17 @@ private static ContainerBuilder AddDbContextOptions<TContext>(this ContainerBuil
...
+             .AddDbContextOptions<StudioManagementSystemDbContextAsync>()
+             .RegisterType<StudioManagementSystemDbContextAsync>()
+             .As<IStudioManagementSystemDbContextAsync>()
+             .InstancePerLifetimeScope();
+         
+         return builder;
+     }
 
     // Example DI for a database context
     // private static ContainerBuilder AddMySpecialDbContext(this ContainerBuilder builder)
```
Comment (albert118) [inline src/server/StudioManagementSystem/DependencyInjectionExtensions.cs:103]: feel free to drop these comments now that we have real examples

**Review by albert118 — CHANGES_REQUESTED** [review]
Looks good overall! Let's just add some logging around those `try-catch` statements to avoid swallowing exceptions silently 😉

**Comment by JaydenL33** [comment]
@albert118 @willyyto, if we are using Supabase for our Database, Auth and API Hosting, do we have any need for our C# backend?

**Comment by albert118** [comment]
> @albert118 @willyyto, if we are using Supabase for our Database, Auth and API Hosting, do we have any need for our C# backend?

I mean it’s already here and quick enough to use. Let’s have a chat Thursday night. Or earlier if you want to catch up on the project

**Comment by albert118** [comment]
## API

image

## Schemas

image

File: src/server/StudioManagementSystem.Core/Entities/MySpecialObject.cs
Comment (albert118) [inline src/server/StudioManagementSystem.Core/Entities/MySpecialObject.cs:1]: removes related

**Pull request merged**